src/main/scala/io/citrine/lolo/learners/ExtraRandomTrees.scala

```diff
@@ -0,0 +1,123 @@
+package io.citrine.lolo.learners
+
+import breeze.stats.distributions.{RandBasis, ThreadLocalRandomGenerator}
+import io.citrine.lolo.bags.Bagger
+import io.citrine.lolo.transformers.FeatureRotator
+import io.citrine.lolo.trees.regression.RegressionTreeLearner
+import io.citrine.lolo.trees.splits.ExtraRandomSplitter
+import io.citrine.lolo.{Learner, TrainingResult}
+import org.apache.commons.math3.random.MersenneTwister
+
+import scala.util.Random
+
+/**
+  * Extremely randomized tree ensemble
+  *
+  * This is based on Geurts, P., Ernst, D. & Wehenkel, L. Extremely randomized trees. Mach Learn 63, 3–42 (2006).
+  * https://doi.org/10.1007/s10994-006-6226-1.
+  *
+  * @param numTrees       number of trees to use (-1 => number of training instances)
+  * @param useJackknife   whether to use jackknife based variance estimates
+  * @param biasLearner    learner to model bias (absolute residual)
+  * @param leafLearner    learner to use at the leaves of the trees
+  * @param subsetStrategy for random feature selection at each split
+  *                       (auto => all features for regression)
+  * @param minLeafInstances minimum number of instances per leave in each tree
+  * @param maxDepth       maximum depth of each tree in the forest (default: unlimited)
+  * @param uncertaintyCalibration whether to empirically recalibrate the predicted uncertainties (default: false)
+  * @param randomlyRotateFeatures whether to randomly rotate real features for each tree in the forest (default: false)
+  */
+case class ExtraRandomTrees(
+                             numTrees: Int = -1,
+                             useJackknife: Boolean = true,
+                             biasLearner: Option[Learner] = None,
+                             leafLearner: Option[Learner] = None,
+                             subsetStrategy: Any = "auto",
+                             minLeafInstances: Int = 1,
+                             maxDepth: Int = Integer.MAX_VALUE,
+                             uncertaintyCalibration: Boolean = false,
```

Comment:
I don't think we can do uncertainty recalibration with bootstrapping disabled, if that ends up being done here.

Reply:
Oh, right, that uses OOB errors.